nucypher/policy/identity.py

```diff
@@ -0,0 +1,389 @@
+        )
+        qr.add_data(bytes(self))
+        qr.print_ascii()
+
+    @classmethod
+    def from_dict(cls, card: Dict):
+        instance = cls(nickname=card.get('nickname'),
+                       verifying_key=card['verifying_key'],
+                       encrypting_key=card['encrypting_key'],
+                       character_flag=card['character'])
+        return instance
+
+    def to_dict(self) -> Dict:
+        payload = dict(
+            nickname=self.__nickname,
+            verifying_key=self.verifying_key,
+            encrypting_key=self.encrypting_key,
+            character=self.__character_flag
+        )
+        return payload
+
+    def to_json(self, as_string: bool = True) -> Union[dict, str]:
+        payload = dict(
+            nickname=self.__nickname.decode(),
+            verifying_key=bytes(self.verifying_key).hex(),
+            encrypting_key=bytes(self.encrypting_key).hex(),
+            character=bytes(self.__character_flag).hex()
+        )
+        if as_string:
+            payload = json.dumps(payload)
+        return payload
+
+    @classmethod
+    def from_character(cls, character: Character, nickname: Optional[str] = None) -> 'Card':
+        flag = getattr(constant_sorrow.constants, character.__class__.__name__.upper())
+        instance = cls(verifying_key=character.public_keys(power_up_class=SigningPower),
+                       encrypting_key=character.public_keys(power_up_class=DecryptingPower),
+                       character_flag=flag,
+                       nickname=nickname)
+        return instance
+
+    #
+    # Card API
+    #
+
+
+    @property
+    def verifying_key(self) -> UmbralPublicKey:
+        return self.__verifying_key
+
+    @property
+    def encrypting_key(self) -> UmbralPublicKey:
+        return self.__encrypting_key
+
+    @property
+    def id(self) -> HexBytes:
+        return self.__hash(self.__payload)
+
+    @property
+    def nickname(self) -> str:
+        if self.__nickname:
+            return self.__nickname.decode()
+
+    def set_nickname(self, nickname: str) -> None:
+        if len(nickname.encode()) > self.__MAX_NICKNAME_SIZE:
+            raise ValueError(f'New nickname exceeds maximum size ({self.__MAX_NICKNAME_SIZE} bytes)')
+        self.__nickname = nickname.encode()
+
+    @nickname.setter
+    def nickname(self, nickname: str) -> None:
+        self.set_nickname(nickname)
+
+    #
+    # Card Storage API
+    #
+
+    @property
+    def filepath(self) -> Path:
+        identifier = f'{self.nickname}{self.__DELIMITER}' if self.__nickname else self.id.hex()
+        filename = f'{identifier}.{self.__FILE_EXTENSION}'
+        filepath = self.card_dir / filename
+        return filepath
+
+    @property
+    def is_saved(self) -> bool:
+        exists = self.filepath.exists()
+        return exists
+
+    def save(self, encoder: Callable = base64.b64encode, overwrite: bool = False) -> Path:
+        if not self.card_dir.exists():
```

Comment:
Is it a consideration that this will eventually (not in this PR of course), use @tuxxy 's Datastore?

Reply:
Hmmm, it might be overkill.  What do you think @tuxxy?